app/src/main/java/com/pmatuki/wowpapers/core/WallpaperApplierImpl.kt

```diff
@@ -7,14 +7,15 @@ import androidx.core.graphics.drawable.toBitmap
 import com.pmatuki.wowpapers.usecases.apply.WallpaperApplier
 import com.pmatuki.wowpapers.usecases.apply.WallpaperApplyResult
 import com.pmatuki.wowpapers.usecases.apply.WallpaperItemHolder
+import dagger.hilt.android.qualifiers.ApplicationContext
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
-import toothpick.InjectConstructor
 import java.io.IOException
+import javax.inject.Inject
 
-@InjectConstructor
-class WallpaperApplierImpl(
-    private val context: Context) : WallpaperApplier<Drawable> {
+class WallpaperApplierImpl @Inject constructor(
```

Comment:
Another think to notice: probably all your `*Impl.kt` classes should be internal. It is a good practice to only expose the public interface and not allow other modules to have access to its implementation.

Reply:
Will do!